### ISSUE TITLE
PME on CUDA and OpenCL returns non-zero potential for any force group without the NonbondedForce

### ISSUE THREAD

**Issue by Lnaden** [pandegroup/openmm#1400]
There is a bug in the `NonbondedForce` when the `NonbondedMethod` is `PME` and `platform` is OpenCL or CUDA. The potential energy for any combination of groups which includes the NonbondedForce is correct, however, all other group combinations that do *not* include the NonbondeForce return a constant non-zero number. Here is a simple test to recreate the issue:

```python
import numpy as np
import simtk.openmm as mm
import simtk.unit as unit

system = mm.System()
Nparticles = 2
positions = np.zeros([Nparticles,3])*unit.nanometer
nbf = mm.NonbondedForce()
cutoff = 3*unit.nanometer
nbf.setCutoffDistance(cutoff)
nbf.setNonbondedMethod(mm.NonbondedForce.PME)
for i in xrange(Nparticles):
    system.addParticle(12)
    nbf.addParticle((-1)**i*0.1, 1, 1)
    positions[i] = (i,0,0)*unit.nanometer
system.addForce(nbf)
pbv = np.zeros([3,3])*unit.nanometer
for i in xrange(3):
    pbv[i,i] = 2.01*cutoff
system.setDefaultPeriodicBoxVectors(*pbv)

timestep = 2*unit.femtoseconds
for platform_index in xrange(mm.Platform.getNumPlatforms()):
    platform = mm.Platform.getPlatform(platform_index)
    platform_name = platform.getName()
    integrator = mm.VerletIntegrator(timestep)
    context = mm.Context(system, integrator, platform)
    context.setPositions(positions)
    print platform_name + " all: ",
    print context.getState(getEnergy=True).getPotentialEnergy()
    print platform_name + " Group 0: ",
    print context.getState(getEnergy=True,groups=2**0).getPotentialEnergy()
    print platform_name + " Group 1: ",
    print context.getState(getEnergy=True,groups=2**1).getPotentialEnergy()
    print platform_name + " Group 2: ",
    print context.getState(getEnergy=True,groups=2**2).getPotentialEnergy()
    print platform_name + " Group 0 and 2: ",
    print context.getState(getEnergy=True,groups=(2**2+2**0)).getPotentialEnergy()
    del context, integrator
```

The Reference and CPU platforms don't have this bug but OpenCL and CUDA do.

**Comment by peastman** [comment]
I just sent a PR with the fix: #1402.  Could you verify that it fixes the problem for you?

**Comment by Lnaden** [comment]
Looks like that takes care of it, Thanks!
